core/src/test/javascript/reportviewer-prompt-spec.js

```diff
@@ -408,12 +413,49 @@ define(["reportviewer/reportviewer-prompt", "reportviewer/reportviewer-logging",
               {value: "test", selected: true},
               {value: "test1", selected: true},
               {value: "test2"}
-            ]
+            ],
+            attributes: {'parameter-render-type': 'list'}
           };
           reportPrompt._applyUserInput("anyname", ["test", "test1", "test2"], parameter);
           expect(parameter.forceUpdate).toBe(true);
         });
 
+        // [BISERVER-14306]
+        // These 2 tests only occur in code when there's a default value for a Date object.  The selection goes through
+        // A loop to first remove the bad date and then adds the new date value. Code was updated to handle the forced
```

Comment:
"A default value for a Date object" -> for a date parameter?
"remove the bad date" -> what is a bad date?

Reply:
In the Report Designer, you can have a calculated default date value by a query. In the javascript, it formats this as a list of date objects (label/value/attributes). With the calculated date default as the first item in the list. So these test cases check this use case specifically.
Bad date not appropriate, updating comment to say the previously selected date